### PR TITLE
chore(autorag+automl): Long names for runs break UI and cannot run

### PR DESCRIPTION
**Pull request by GAUNSD**
## Issues

- https://redhat.atlassian.net/browse/RHOAIENG-57323
- https://redhat.atlassian.net/browse/RHOAIENG-57335

## Description

This PR addresses issues with extra long names being provided for both AutoML and AutoRAG runs.
* `<Truncate />` has been added to each Result and Configure pages to avoid large overflow
* The BFF is now restricting the max length to be `250`. (Aligning the limit here to the reasonable limit found for project names in ODH)
* The zod schema for UI has been aligned to the new `250` limit

image
image

## How Has This Been Tested?

* Manual testing
* New test cases added

## Test Impact

* 🆕 Tests cases for zod schema
* 🆕 Tests cases for BFF length limit

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Summary by CodeRabbit

* **Bug Fixes & Improvements**
  * Enforced a 250-character maximum for display names system-wide; overly long names are rejected with a validation error.
  * Long display names are truncated in the UI (breadcrumbs, headings, result/subtext) to keep layouts stable and readable.

* **Tests**
  * Added unit and contract tests covering 250- and 251-character boundary cases (including multi-byte characters).

### CODE REVIEW

File: packages/automl/api/openapi/automl.yaml [hunk 1]
```diff
@@ -1133,6 +1133,7 @@ components:
         display_name:
           type: string
           minLength: 1
+          maxLength: 250
           description: Human-readable name for the run
```
[the remark this answers is not in the document]
Comment (GAUNSD) [reply packages/automl/api/openapi/automl.yaml]: Pushed a fix: https://github.com/opendatahub-io/odh-dashboard/pull/7156/commits/7caca3a1c7c36562890f2235df79ec1f465752f8

**Review by MatthewAThompson — COMMENTED** [review]
Thanks @GAUNSD, code mostly looks good, just one comment:
